### PR TITLE
added flag rescueChangesNotSendForReview

### PR DESCRIPTION
**Pull request by devops-ninja2021**
**Task name**: [GooglePlayReleaseV4 sends changes to review even if option changesNotSentForReview is set to false](https://github.com/microsoft/google-play-vsts-extension/issues/324#)

**Description**: added the new flag rescueChangesNoSentForReview. If the changesNoSentForReview fails the flag is inverted and retried to commit the edit.

**Documentation changes required:** Y Description of new flag rescueChangesNoSentForReview

**Added unit tests:** N 

**Attached related issue:** (Y/N) #324

**Checklist**:
- [X ] Task version was bumped - please check [instruction](https://github.com/microsoft/google-play-vsts-extension/tree/master/docs/taskversionbumping.md) how to do it
- [ X] Checked that applied changes work as expected

### CODE REVIEW

File: vsts-extension-google-play.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
     "manifestVersion": 1.0,
     "id": "google-play",
     "name": "Google Play",
-    "version": "4.201.0",
+    "version": "4.201.1",
```
Comment (anatolybolshakov) [inline vsts-extension-google-play.json:5]: Could you please bump minor task version to the [current sprint](https://whatsprintis.it/) - 204?

Comment (anatolybolshakov) [inline vsts-extension-google-play.json:5]: Could you please bump minor task version to the [current sprint](https://whatsprintis.it/) - 204?

File: Tasks/GooglePlayReleaseV4/task.json [hunk 2]
```diff
@@ -237,6 +237,14 @@
             "helpMarkDown": "Select this option to send changes for review in GooglePlay Console. If changes are already sent for review automatically, you shouldn't select this option. [More info](https://developers.google.com/android-publisher/api-ref/rest/v3/edits/commit#query-parameters).",
             "groupName": "advanced"
         },
```
Comment (anatolybolshakov) [inline Tasks/GooglePlayReleaseV4/task.json:239]: Please bump relevant task version - see [steps](https://github.com/microsoft/google-play-vsts-extension/tree/master/docs/taskversionbumping.md) for more details

Comment (devops-ninja2021) [reply Tasks/GooglePlayReleaseV4/task.json:239]: Bumped the minor version to 204

File: Tasks/GooglePlayReleaseV4/task.json [hunk 3]
```diff
@@ -256,6 +256,14 @@
...
+        {
+            "name": "rescueChangesNotSentForReview",
+            "type": "boolean",
+            "label": "Retry commit if changesNotSentForReview failed",
+            "defaultValue": false,
+            "helpMarkDown": "Retry commit if changesNotSentForReview failed",
+            "groupName": "advanced"
+        },
```
Comment (ismayilov-ismayil) [inline Tasks/GooglePlayReleaseV4/task.json:266]: Is this worth to add visibility rule by the changesNotSentForReview as ```"visibleRule": "changesNotSentForReview = true"```  due to they are dependent?

**Review by ismayilov-ismayil — COMMENTED** [review]
Hi @devops-ninja2021, thanks for your contribution.
I tested your changes, and it works as expected. 
Just small comment for new field

File: Tasks/GooglePlayReleaseV4/task.json (hunk 3 above)
Comment (devops-ninja2021) [reply Tasks/GooglePlayReleaseV4/task.json:267]: Hi @ismayilov-ismayil,
you are right. Should be ok if it field is only displayed when changesNotSentForReview = true.

**Review by ismayilov-ismayil — APPROVED** [review]

**Comment by ismayilov-ismayil** [comment]
Hi @devops-ninja2021 
Could you please bump task version as well. We already have 4.220.0, in this case you change will 4.220.1

**Comment by ismayilov-ismayil** [comment]
@devops-ninja2021 thanks for changes. One more suggestion: Could you please move strings to task.json as localization and call it by tl.loc method

**Comment by devops-ninja2021** [comment]
> @devops-ninja2021 thanks for changes. One more suggestion: Could you please move strings to task.json as localization and call it by tl.loc method

Hi @ismayilov-ismayil,
I'm not familiar with the localization mechanism of the extension. I did some localization replacements in task.loc.json. If this isn't the point you like to get it changed, I would please you to adjust it by yourself. I will have a look on it then for future changes.
Thanks.

**Review by kirill-ivlev — APPROVED** [review]

**Review by tkasparek — APPROVED** [review]

**Pull request merged**